### PR TITLE
Slices skeleton for new update

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #227 
Added slices skeleton for the new update process, each slice will process different files, `MetadataChecksumMavenSlice` will initiate repository update. Also I'm planning to rename `MetadataChecksumMavenSlice` to `UpdateMavenSlice` or `DeployMavenSlice` as soon as we move to the new slices and `UpdateMavenSlice` will be not needed.

### CODE REVIEW

File: src/main/java/com/artipie/maven/http/MetadataMavenSlice.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.reactivestreams.Publisher;
+
+/**
+ * This slice accepts PUT requests with maven-metadata.xml, reads `latest` version from the
+ * file and saves it to the temp location adding version before the filename:
+ * `.upload/${package_name}/${version}/maven-metadata.xml`.
+ * @since 0.8
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class MetadataMavenSlice implements Slice {
```
Comment (g4s8) [inline src/main/java/com/artipie/maven/http/MetadataMavenSlice.java:41]: @olenagerasimova if this slice is for PUT requests, I'd add some prefix to it like `PutMetadataSlice` or `UpdateMetadaSlice` or similar

File: src/main/java/com/artipie/maven/http/MetadataChecksumMavenSlice.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * This slice accepts PUT requests with maven-metadata.xml checksums, maven-metadata.xml.sha1 and
+ * maven-metadata.xml.md5, picks up corresponding maven-metadata.xml from the package upload temp
+ * location and saves the checksum. If two checksums (sha1 and md5) are present in the package
+ * upload location, this slice initiate repository update.
+ * @since 0.8
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class MetadataChecksumMavenSlice implements Slice {
```
Comment (g4s8) [inline src/main/java/com/artipie/maven/http/MetadataChecksumMavenSlice.java:42]: @olenagerasimova and same here: if it's only for PUT requests, I'd add a prefix

File: src/main/java/com/artipie/maven/http/MetadataChecksumMavenSlice.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.commons.lang3.NotImplementedException;
+import org.reactivestreams.Publisher;
+
+/**
+ * This slice accepts PUT requests with maven-metadata.xml checksums, maven-metadata.xml.sha1 and
+ * maven-metadata.xml.md5, picks up corresponding maven-metadata.xml from the package upload temp
```
Comment (g4s8) [inline src/main/java/com/artipie/maven/http/MetadataChecksumMavenSlice.java:36]: @olenagerasimova maven also support sha256

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova please check my comments

**Comment by olenagerasimova** [comment]
@g4s8 thanks, renamed the slices and corrected javadoc

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**